crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -466,10 +467,20 @@ impl AuthorityPerEpochStore {
 
         let signature_verifier =
             SignatureVerifier::new(committee.clone(), signature_verifier_metrics);
-        for (_, v) in oauth_provider_jwk.iter() {
-            signature_verifier.insert_oauth_jwk(v);
+        for ((kid, iss), v) in oauth_provider_jwk.iter() {
+            signature_verifier.insert_oauth_jwk(v, kid.to_string(), iss.to_string());
         }
 
+        let zklogin_env = match protocol_config.zklogin_use_secure_vk() {
+            true => ZkLoginEnv::Prod,
+            false => ZkLoginEnv::Test,
+        };
+
+        signature_verifier.set_config(
```

Comment:
can we simply pass the protocol config to SignatureVerifier so it can consult it directly?

Reply:
revised